docs/how-to-guides/block-tutorial/extending-the-query-loop-block.md

```diff
@@ -0,0 +1,195 @@
+```
+
+Of course, you'll be responsible for implementing the logic of your control (you might want to take a look at [`@wordpress/components`](https://www.npmjs.com/package/@wordpress/components) to make your controls fit seamlessly within the Gutenberg UI). Any extra parameter you assign within the `query` object inside the blocks attributes can be used to create a custom query according to your needs, with a little extra effort.
+
+Currently, you'll likely have to implement slightly different paths to make the query behave correctly both on the front-end side (i.e. on the end user's side) and to show the correct preview on the editor side.
+
+```js
+{
+	/** ...variation properties */
+	attributes: {
+		/** ...variation attributes */
+		query: {
+			/** ...more query settings if needed */
+			postType: 'book',
+			/** Our custom query parameter */
+			bookAuthor: 'J. R. R. Tolkien'
+		}
+	}
+}
+```
+
+### Making your custom query work on the front-end side
+
+The Query Loop block functions mainly through the Post Template block which receives the attributes and builds the query from there. Other first-class children of the Query Loop block (such as the Pagination block) behave in the same way. They build their query and then expose the result via the filter [`query_loop_block_query_vars`](https://developer.wordpress.org/reference/hooks/query_loop_block_query_vars/).
+
+You can hook into that filter and modify your query accordingly. Just make sure you don't cause side-effects to other Query Loop blocks by at least checking that you apply the filter only to your variation!
+
+```php
+if( 'my-plugin/books-list' === $block[ 'attrs' ][ 'namespace' ] ) {
+	add_filter(
+		'query_loop_block_query_vars',
+		function( $query ) {
+			/** You can read your block custom query parameters here and build your query */
+		},
+	);
+}
+```
+
+### Making your custom query work on the editor side
+
+To finish up our custom variation, we might want the editor to react to changes in our custom query and display an appropriate preview accordingly. This is not required for a functioning block, but it enhables a fully integrated user experience for the consumers of your block.
+
+The Query Loop block fetches its posts to show the preview using the [WordPress REST API](https://developer.wordpress.org/rest-api/). Any extra parameter added to the `query` object will be passed as a query argument to the API. This means that these extra parameters should be either supported by the REST API, or be handled by custom filters such as the [`rest_{$this->post_type}_query`](https://developer.wordpress.org/reference/hooks/rest_this-post_type_query/) filter which allows you to hook into any API request for your custom post type. Like so:
+
+```php
+add_filter(
+	'rest_book_query',
+	function( $args, $request ) {
+		/** We can access our custom parameters from here */
+		$book_author = $request->get_param( 'bookAuthor' );
```

Comment:
Does it make sense to add just a bit more code in this specific example?

Reply:
Mm, I don't know. Again, I didn't want to be too prescriptive and specific. The way this query is modified is highly dependant on your plugin, after all. Personally, I'd rather leave it open, maybe specifying that they need to return the new query arguments? Though it feels redundant with the docs on the hook, what do you think?

Reply:
I'll trust you on this since you have made lots of exploration for the extension of the block.